### PR TITLE
fix: general ui issues

### LINKED ISSUE #235: Error when trying to swap to pair that doesn't have pool yet
**Problem**
When typing "from" value, it throws and unhandled error.

**Solution**
We should in the button "No liquidity for ETH/DAI"

### LINKED ISSUE #251: Text typos and grammar
"Click the button below to mint 0.5 ETH to your wallet."

**Should be updated to:**

"Click the button below to receive 0.5 ETH to your wallet."

---

"You may have no open positions"

**Should be updated to:**

"You do not have any open positions"

### LINKED ISSUE #252: Odd wallet remainder after ETH Swap to Dai
1. New private browser window
2. Visit https://fuellabs.github.io/swayswap/swap
3. Faucet Eth
4. Click "Max" for Eth
5. Swap for Dai
6. The remaining ETH balance in the wallet should be zero, but is instead: 0.000000001 ETH

I think there might be either a fee, calculation or display issue here.

### LINKED ISSUE #253: No faucet button visible in mobile view
signal-2022-06-02-174250

### LINKED ISSUE #254: Add liquidity shows position as 100% when the values is empty
Screen Shot 2022-06-02 at 10 18 25 AM

### PR DESCRIPTION
**Pull request by pedronauck**
✨ Fix #235
✨ Fix #251 
✨ Fix #252
✨ Fix #253

### CODE REVIEW

File: packages/app/src/systems/Pool/hooks/usePreviewAddLiquidity.ts [hunk 1]
```diff
@@ -18,11 +18,15 @@ export interface UsePreviewAddLiquidityProps {
 
 export function usePreviewAddLiquidity({ fromInput, poolInfo }: UsePreviewAddLiquidityProps) {
   const { totalLiquidity, poolTokensNum } = useUserPositions();
+
   const liquidityFactor = (fromInput.amount || ZERO) * toBigInt(totalLiquidity);
   const previewTokens = divideFnValidOnly(liquidityFactor, totalLiquidity);
   const nextTotalTokenSupply = previewTokens + toNumber(poolInfo?.lp_token_supply || BigInt(0));
-  const nextCurrentPoolShare =
-    divideFnValidOnly(BigInt(previewTokens) + poolTokensNum, BigInt(nextTotalTokenSupply)) || 1;
```
Comment (luizstacio) [inline packages/app/src/systems/Pool/hooks/usePreviewAddLiquidity.ts:25]: The `|| 1` here is used on liquidity creation to show 100% as the user's current position. Did you test it in this state?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Core/utils/math.ts [hunk 2]
```diff
@@ -61,7 +61,9 @@ export function parseToFormattedNumber(
   if (typeof value === 'number') {
     val = BigInt(Math.trunc(value));
   }
-  return ethers.commify(toFixed(formatUnits(val, precision), FIXED_UNITS));
+  const result = ethers.commify(toFixed(formatUnits(val, precision), FIXED_UNITS));
+  const decimal = new Decimal(result.replaceAll(',', ''));
+  return decimal.lessThanOrEqualTo('0.0001') ? '0' : result;
```
Comment (luizstacio) [inline packages/app/src/systems/Core/utils/math.ts:66]: Why we need this here?

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/systems/Core/utils/math.ts (hunk 2 above)
Comment (pedronauck) [reply packages/app/src/systems/Core/utils/math.ts:66]: Yes, is returning 100% when the user has 100% of the pool and 0 when he have nothing.

File: packages/app/src/systems/Pool/hooks/usePreviewAddLiquidity.ts (hunk 1 above)
Comment (pedronauck) [reply packages/app/src/systems/Pool/hooks/usePreviewAddLiquidity.ts:25]: Yes, is returning 100% when the user has 100% of the pool and 0 when he has nothing.

File: packages/app/src/systems/Core/utils/math.ts (hunk 2 above)
Comment (pedronauck) [reply packages/app/src/systems/Core/utils/math.ts:66]: what do you recommend doing in this case @luizstacio?

Comment (luizstacio) [reply packages/app/src/systems/Core/utils/math.ts:66]: let's remove the "hack" implementation and address it on the gas fee in tx. For let the issue open.

Comment (pedronauck) [reply packages/app/src/systems/Core/utils/math.ts]: ✅

**Comment by luizstacio** [comment]
`Your share of current pool:` should be the current position + the new position to be added

In the case of creating it would be 100% but now is showing 0
Screen Shot 2022-06-07 at 4 20 57 PM

**Comment by pedronauck** [comment]
> `Your share of current pool:` should be the current position + the new position to be added

But this should be 100% just after I create the pool, no?! Because if there's no pool created, my current share should be nothing, and then after I create my current share is 100%. That's what's happening now.

**Comment by luizstacio** [comment]
> > `Your share of current pool:` should be the current position + the new position to be added
> 
> @luizstacio But this should be 100% just after I create the pool, no?! Because if there's no pool created, my current share should be nothing, and then after I create my current share is 100%. That's what's happening now.

No as if also does the `+ new position`

**Review by luizstacio — DISMISSED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**